packages/valory/skills/apy_estimation_abci/io_/load.py

```diff
@@ -0,0 +1,90 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2021-2022 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains all the loading operations of the APY behaviour."""
+
+from typing import Callable, Dict, Optional
+
+import joblib
+import pandas as pd
+
+from packages.valory.skills.abstract_round_abci.io_.load import AbstractLoader
+from packages.valory.skills.abstract_round_abci.io_.load import Loader as BaseLoader
+from packages.valory.skills.apy_estimation_abci.io_.store import (
+    CustomObjectType,
+    ExtendedSupportedFiletype,
+    NativelySupportedSingleObjectType,
+    SupportedSingleObjectType,
+)
+
+
+CustomLoaderType = Optional[Callable[[str], CustomObjectType]]
+SupportedLoaderType = Callable[[str], SupportedSingleObjectType]
+
+
+class CSVLoader(AbstractLoader):
+    """A csv files Loader."""
+
+    def load_single_file(self, path: str) -> NativelySupportedSingleObjectType:
+        """Read a pandas dataframe from a csv file.
+
+        :param path: the path of the csv.
+        :return: the pandas dataframe.
+        """
+        try:
+            return pd.read_csv(path)
+        except FileNotFoundError as e:  # pragma: no cover
+            raise IOError(f"File {path} was not found!") from e
+        except pd.errors.EmptyDataError as e:  # pragma: no cover
+            raise IOError("The provided csv was empty!") from e
+
+
+class ForecasterLoader(AbstractLoader):
+    """A `pmdarima` forecaster loader."""
+
+    def load_single_file(self, path: str) -> NativelySupportedSingleObjectType:
+        """Load a `pmdarima` forecaster.
+
+        :param path: path to store the forecaster.
+        :return: a `pmdarima.pipeline.Pipeline`.
+        """
+        try:
+            return joblib.load(path)
+        except (NotADirectoryError, FileNotFoundError) as e:  # pragma: no cover
+            raise IOError(f"Could not detect {path}!") from e
```

Comment:
quite frankly just excessive code. This can all be deleted, and `pd.read_csv` and `joblib.load` used where these classes and their single method (!) are referenced. Classes with a single method that do not reference `self` should not exist. Since this is a refactor now would be the perfect time.

Bonus: we get rid of some `# pragma: no cover`

Reply:
No, we cannot just use `pd.read_csv` and `joblib.load`. We want the `IPFSInteract` class to be able to call a `store` or `load` method and look for a single exception, `IOError` (ideally should be a custom exception), and a relevant message. We need these methods in order to filter all the exceptions and raise the relevant ones.

Moreover, these classes do not have a single method. They **override** a single method.

Reply:
thanks for clarifying. Yes custom exception for intercepting makes sense, but not obvious since that is not what is going on